### PR TITLE
pyproject.toml: revert "dev" dependencies and requirements file

### PR DESCRIPTION
**Pull request by Exirel**
### Description

As previously discussed, using `pyproject.toml` for Sopel development dependencies broke my workflow, because it removed my ability to upgrade simply and easily all my dev tools for Sopel.

Moreover, the purpose of extra dependencies in `pyproject.toml` (same for `setup.cfg`) is to allow third party developers to install Sopel with tools to develop for Sopel, i.e. plugins, tools, etc. and not Sopel itself.

I used the `requirements.txt` file instead of the `dev-requirements.txt`, since there is no other purpose for "requirements" than development.

And by using a requirement file, I was able to upgrade all my tools properly in one short command, and to discover some minor errors with `mypy`! That I'll need to fix sooner or later... preferably before Sopel 8 release.

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,10 +66,12 @@ Assuming you are using standard tools:
 * Fork Sopel's repository, and clone your fork locally.
 * Activate your virtualenv, and `cd` into your clone's folder.
 * Ensure you have the latest version of `pip`, and install `wheel`.
-* Install Sopel from source as an editable install using the `dev` extra.
+* Install Sopel from source as an editable install, and install the dev
+  depencendies using the `requirements.txt` file:
```
Comment (dgw) [inline CONTRIBUTING.md:70]:
```suggestion
  dependencies using the `requirements.txt` file:
```

I also think we should not use the de facto standard name "requirements.txt" for non-runtime dependencies. The old "dev-requirements.txt" was fine, but something shorter would also work. "devdeps.txt" (or with a hyphen) maybe?

**Review by dgw — COMMENTED** [review]
Just a typo and alternate suggestion for requirement file name. I'll have to tweak the configuration for docs preview on Netlify if we use something other than "dev-requirements.txt".

**Comment by Exirel** [comment]
I went back to `dev-requirements.txt`, since I don't have a strong opinion about its name and that's what there was before.

**Comment by dgw** [comment]
Needs `requirements.txt` -> `dev-requirements.txt` in CONTRIBUTING.md

**Review by dgw — APPROVED** [review]

**Comment by Exirel** [comment]
Oh, this got merged without a proper squash of the commits...

Well, it's OK!

**Comment by dgw** [comment]
Uh… Whoops. 🙈 "It'll be fine." — _KatherineOfSky, frequently_

**Pull request merged**